File: atcoder/abs/cpp/abc085_c.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, y;

    cin >> n;
    cin >> y;

    for (int yukichi = 0; yukichi <= n; yukichi++)
        for (int higuchi = 0; higuchi <= n - yukichi; higuchi++) {
            int noguchi = n - yukichi - higuchi;
            if ((yukichi * 10000 + higuchi * 5000 + noguchi * 1000) == y) {
                printf("%d %d %d\n", yukichi, higuchi, noguchi);
                return 0;
            }
        }

    printf("-1 -1 -1");
}

File: atcoder/abs/perl5/abc085_c.pl
use strict;
use warnings;
use feature 'say';
sub get { my $input = <>; chomp($input); $input }

my ($N, $Y) = split " ", get;

for my $yukichi (0..$N) {
    for my $higuchi (0..$N-$yukichi) {
        my $noguchi = $N - $yukichi - $higuchi;
        if (($yukichi * 10000 + $higuchi * 5000 + $noguchi * 1000) == $Y) {
            say "$yukichi $higuchi $noguchi";
            exit;
        }
    }
}

say "-1 -1 -1";